cpp/src/copying/contiguous_split.cu

```diff
@@ -1024,15 +1025,18 @@ struct packed_split_indices_and_src_buf_info {
       reinterpret_cast<size_type*>(reinterpret_cast<uint8_t*>(d_indices_and_source_info.data()) +
                                    indices_size + src_buf_info_size);
 
-    CUDF_CUDA_TRY(cudaMemcpyAsync(
-      d_indices, h_indices, indices_size + src_buf_info_size, cudaMemcpyDefault, stream.value()));
+    detail::cuda_memcpy_async<uint8_t>(
+      device_span<uint8_t>{static_cast<uint8_t*>(d_indices_and_source_info.data()),
+                           h_indices_and_source_info.size()},
+      h_indices_and_source_info,
+      stream);
```

Comment:
Similarly, can we use this, with `*_inidices_and_source_info` declared to be vectors of `uint8_t`?
```suggestion
    detail::cuda_memcpy_async<uint8_t>(
      d_indices_and_source_info,
      h_indices_and_source_info,
      stream);
```

Reply:
can't implicitly cast a device_buffer to device_span

Reply:
So can we make `d_indices_and_source_info` a `device_uvector<int8_t>` (or `char`)?

Reply:
The data is cast into different types:
```
    d_indices      = reinterpret_cast<size_type*>(d_indices_and_source_info.data());
    d_src_buf_info = reinterpret_cast<src_buf_info*>(
      reinterpret_cast<uint8_t*>(d_indices_and_source_info.data()) + indices_size);
```
I feel like a typeless container fits the best here.